packages/@sanity/base/src/datastores/authToken/token.ts

```diff
@@ -0,0 +1,42 @@
+import {versionedClient} from '../../client/versionedClient'
+
+// Project ID is part of the localStorage key so that different projects can store their separate tokens, and it's easier to do book keeping.
+const getLSKey = (projectId: string) => {
+  if (!projectId) {
+    throw new Error('Invalid project id')
+  }
+  return `__studio_auth_token_${projectId}`
+}
+
+export const saveToken = ({token, projectId}: {token: string; projectId: string}): void => {
+  window.localStorage.setItem(
+    getLSKey(projectId),
+    JSON.stringify({token, time: new Date().toISOString()})
+  )
+}
+
+export const deleteToken = (projectId: string): void => {
+  window.localStorage.removeItem(getLSKey(projectId))
+}
+
+export const getToken = (projectId: string): string | null => {
+  if (!window.localStorage) {
+    return null
+  }
+  const item = window.localStorage.getItem(getLSKey(projectId))
+  if (item) {
+    const {token}: {token: string} = JSON.parse(item)
+    if (token) {
+      return token
+    }
+  }
+  return null
+}
```

Comment:
ditto. i've worked at places that have the security settings on machines set really high and it throw if you touch localStorage. [this](https://gist.github.com/ricokahler/a11513365fd83d31e8f3ff6ca481343f) is what i've done in the past. probably not totally applicable here but just in case

Reply:
i don't think we can shim around local storage being unavailable in terms of cross-window comm. maybe worth considering another web API

Reply:
I think we could live without the cross-window stuff in that case if we can offer once-a-session-in-window logins for those with extreme settings. I suggest we don't spend time on it now though.

I have however made the LS methods more resilient for the kind of problems @rexxars mentions.